crates/relayer-framework/src/tests/relayer_mock/base/impls/relay.rs

```diff
@@ -0,0 +1,157 @@
+use crate::tests::relayer_mock::base::types::runtime::MockRuntimeContext;
+use crate::tests::relayer_mock::contexts::chain::MockChainContext;
+use crate::tests::relayer_mock::contexts::relay::MockRelayContext;
+
+impl OfaRelayTypes for MockRelayContext {
+    type Preset = MinimalPreset;
+
+    type Error = Error;
+
+    type Runtime = MockRuntimeContext;
+
+    type Packet = PacketKey;
+
+    type SrcChain = MockChainContext;
+
+    type DstChain = MockChainContext;
+}
+
+#[async_trait]
+impl OfaBaseRelay for MockRelayContext {
+    fn mismatch_ibc_events_count_error(expected: usize, actual: usize) -> Self::Error {
+        Error::mismatch_error(expected, actual)
+    }
+
+    fn packet_src_channel_id(
+        packet: &Self::Packet,
+    ) -> &<Self::SrcChain as OfaChainTypes>::ChannelId {
+        &packet.channel_id
+    }
+
+    fn packet_src_port(packet: &Self::Packet) -> &<Self::SrcChain as OfaChainTypes>::PortId {
+        &packet.channel_id
+    }
+
+    fn packet_dst_port(packet: &Self::Packet) -> &<Self::DstChain as OfaChainTypes>::PortId {
+        &packet.port_id
+    }
+
+    fn packet_dst_channel_id(
+        packet: &Self::Packet,
+    ) -> &<Self::DstChain as OfaChainTypes>::ChannelId {
+        &packet.channel_id
+    }
+
+    fn packet_sequence(packet: &Self::Packet) -> &<Self::SrcChain as OfaChainTypes>::Sequence {
+        &packet.sequence
+    }
+
+    fn packet_timeout_height(
+        packet: &Self::Packet,
+    ) -> Option<&<Self::DstChain as OfaChainTypes>::Height> {
+        Some(&packet.timeout_height)
+    }
+
+    fn packet_timeout_timestamp(
+        packet: &Self::Packet,
+    ) -> &<Self::DstChain as OfaChainTypes>::Timestamp {
+        &packet.timeout_timestamp
+    }
+
+    fn runtime(&self) -> &OfaRuntimeContext<Self::Runtime> {
+        &self.runtime
+    }
+
+    fn src_client_id(&self) -> &<Self::SrcChain as OfaChainTypes>::ClientId {
+        self.src_to_dst_client()
+    }
+
+    fn src_chain(&self) -> &OfaChainWrapper<Self::SrcChain> {
+        &self.src_chain
+    }
+
+    fn dst_client_id(&self) -> &<Self::DstChain as OfaChainTypes>::ClientId {
+        self.dst_to_src_client()
+    }
+
+    fn dst_chain(&self) -> &OfaChainWrapper<Self::DstChain> {
+        &self.dst_chain
+    }
+
+    async fn build_src_update_client_messages(
+        &self,
+        height: &<Self::DstChain as OfaChainTypes>::Height,
+    ) -> Result<Vec<<Self::SrcChain as OfaChainTypes>::Message>, Self::Error> {
+        let state = self.src_chain().chain.state();
+        Ok(vec![MockMessage::UpdateClient(
+            self.src_client_id().to_string(),
+            height.clone(),
+            state,
+        )])
+    }
+
+    async fn build_dst_update_client_messages(
+        &self,
+        height: &<Self::SrcChain as OfaChainTypes>::Height,
+    ) -> Result<Vec<<Self::DstChain as OfaChainTypes>::Message>, Self::Error> {
+        let state = self.dst_chain().chain.state();
```

Comment:
We should give back the state at the specific height. Although since our mock chain do not have a state at every height, it should give back the state that is at the earliest after the specified height. If all heights are before the given height, it should return error.